### PR TITLE
Fix 4058 : restore from recycle bin when item has multiple relations

### PR DESCRIPTION
**Pull request by dawoe**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes: #4058 

### Description

Before this fix the relationtype alias was not passed to the API. So when you have multiple relations to a item in the recycle bin this could not be restored, because all relations were returned by the api. 

I tested this by installing my nexu package and deleting a item that in use. Then the restore option breaks.

This was because the querystring parameter of umbrequesthelper.getApiUrl was passed as an array ( https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Web.UI.Client/src/common/services/umbrequesthelper.service.js#L83 )

Because of this the method umbRequestHelper.dictionaryToQueryString returned only the chilid parameter and not the alias
https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Web.UI.Client/src/common/services/umbrequesthelper.service.js#L44

Fixed this by passing the parameters as json object.

I also needed to set the correct relation type alias when restoring media.

### CODE REVIEW

**Comment by emmaburstow** [comment]
Hi @dawoe!

Thanks for the contribution. You know what'll happen next. Happy new year!

Em

File: src/Umbraco.Web.UI.Client/src/common/resources/relation.resource.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function relationResource($q, $http, umbRequestHelper) {
                     umbRequestHelper.getApiUrl(
                         "relationApiBaseUrl",
                         "GetByChildId",
-                        [{ childId: id, relationTypeAlias: alias }])),
+                        { childId: id, relationTypeAlias: alias })),
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/common/resources/relation.resource.js]: I'm not too sure about the syntax @dawoe - why do we need to remove the `[` / `]`?

Comment (nul800sebastiaan) [reply src/Umbraco.Web.UI.Client/src/common/resources/relation.resource.js]: Didn't mean to request changes, it was just a question ;-)

Comment (dawoe) [reply src/Umbraco.Web.UI.Client/src/common/resources/relation.resource.js]: If you pass it as an array the logic to turn it in to a querystring works different and parameters were not passed correctly to the api.

When you pass it as an object it works correctly.

**Comment by dawoe** [comment]
Merged in latest dev-v7 to solve merge conflicts

**Comment by nul800sebastiaan** [comment]
Thanks!

**Pull request merged**